src/axom/slic/streams/SynchronizedStream.cpp

```diff
@@ -57,15 +61,70 @@ SynchronizedStream::SynchronizedStream(std::ostream* stream,
   : m_comm(comm)
   , m_cache(new MessageCache)
   , m_stream(stream)
+  , m_file_name()
+  , m_isOstreamOwnedBySLIC(false)
+  , m_opened(false)
 {
   this->setFormatString(format);
 }
 
+//------------------------------------------------------------------------------
+SynchronizedStream::SynchronizedStream(const std::string stream, MPI_Comm comm)
+{
```

Comment:
(very) minor: The initialization of `m_cache` and `m_comm` are common to all three and can be initialized using a member initialization list, i.e.:
```cpp
SynchronizedStream::SynchronizedStream(const std::string stream, MPI_Comm comm) 
     : m_comm(comm)
     , m_cache(new MessageCache)  
{
 ...
}
```